### PR TITLE
fix(api)!: correctly deal with number before :tab

### PR DESCRIPTION
**Pull request by zeertzjq**
Now `nvim_parse_cmd` and `nvim_create_user_command` use a `tab` value which is the same as the number passed before `:tab` modifier instead of the number plus 1, and `tab` value is -1 if `:tab` modifier is not used.

### CODE REVIEW

File: src/nvim/api/command.c [hunk 1]
```diff
@@ -557,10 +557,12 @@ String nvim_cmd(uint64_t channel_id, Dict(cmd) *cmd, Dict(cmd_opts) *opts, Error
     }
 
     if (HAS_KEY(mods.tab)) {
-      if (mods.tab.type != kObjectTypeInteger || mods.tab.data.integer < 0) {
-        VALIDATION_ERROR("'mods.tab' must be a non-negative Integer");
+      if (mods.tab.type != kObjectTypeInteger) {
+        VALIDATION_ERROR("'mods.tab' must be a Integer");
```
Comment (lewis6991) [inline src/nvim/api/command.c:561]:
```suggestion
        VALIDATION_ERROR("'mods.tab' must be an Integer");
```

Comment (zeertzjq) [reply src/nvim/api/command.c]: I used the same as the one below, so I'll change both.

**Comment by lewis6991** [comment]
Shouldn't `tab` be `nil` if not used?

**Comment by zeertzjq** [comment]
> Shouldn't `tab` be `nil` if not used?

Not sure about this. This is inconsistent with `verbose`, but these two modifiers have very different meanings, so this may make sense.

**Comment by lewis6991** [comment]
I beg to ask why does `verbose` do this? If there isn't a good reason, that too should default to nil.

**Comment by lewis6991** [comment]
Ah I see. `-1` is begin used as `false` for integer values. Yeah, then maybe this makes sense.

**Pull request merged**